apps/SelfDemo/app/src/main/java/com/selfxyz/demo/MainActivity.kt

```diff
@@ -0,0 +1,269 @@
+
+    // This would be given to the self sdk from the app (ie minipage and stored in secure storage)
+    private val appSecret = "your-secret-key-here"
+
+    companion object {
+        private const val TAG = "MainActivity"
+    }
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        enableEdgeToEdge()
+
+        // Initialize NFC
+        nfcAdapter = NfcAdapter.getDefaultAdapter(this)
+        if (nfcAdapter == null) {
+            Toast.makeText(this, "NFC is not available on this device", Toast.LENGTH_LONG).show()
+        }
+
+        // Create pending intent for NFC
+        val intent = Intent(this, javaClass).apply {
+            addFlags(Intent.FLAG_ACTIVITY_SINGLE_TOP)
+        }
+        pendingIntent = PendingIntent.getActivity(
+            this, 0, intent,
+            PendingIntent.FLAG_UPDATE_CURRENT or PendingIntent.FLAG_MUTABLE
+        )
+
+        setContent {
+            SelfDemoTheme {
+                Scaffold(modifier = Modifier.fillMaxSize()) { innerPadding ->
+                    WebViewScreen(
+                        modifier = Modifier.padding(innerPadding),
+                        onWebViewCreated = { webView ->
+                            this.webView = webView
+                            setupWebView()
+                        }
+                    )
+                }
+            }
+        }
+    }
+
+    private fun setupWebView() {
+        webView.settings.apply {
+            javaScriptEnabled = true
+            domStorageEnabled = true
+            allowFileAccess = true
+            allowContentAccess = true
+            mixedContentMode = android.webkit.WebSettings.MIXED_CONTENT_ALWAYS_ALLOW
+        }
+
+        // Add JavaScript interface
+        webView.addJavascriptInterface(WebAppInterface(), "AndroidApp")
+
+        webView.webViewClient = object : WebViewClient() {
+            override fun onPageFinished(view: WebView?, url: String?) {
+                super.onPageFinished(view, url)
+                // Inject the secret securely after page loads
+                injectSecret()
+            }
+        }
+
+        // Load the embedded.self.xyz website
+        webView.loadUrl("https://embedded.self.xyz")
+    }
+
+    private fun injectSecret() {
+        // TODO before sending secret to the app can we authenticate that it is legit?
+        // OR maybe instead of injecting immediately it would be better to only request the secret when needed?
+        val secretHash = appSecret
+        val script = """
+
+            if (window.selfApp) {
+                window.selfApp.setSecret('$secretHash');
+            } else {
+                // Fallback: store in a global variable if the web app isn't ready
+                window.selfAppSecret = '$secretHash';
+            }
+        """.trimIndent()
+
+        webView.post {
+            webView.evaluateJavascript(script, null)
+        }
+    }
```

Comment:
this doesnt feel nice. instead we can define a method like Android.getPrivateKey() that is callable from js side. 

out koktlin module will either A) require the caller to pass the private key in on init or B) require a function to be passed in that when called returns the private key (probably after asking for it from secure enclave)